src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -110,6 +116,58 @@ public String linkAllAccountsWithBots(Pageable pageable) {
         throw new FriendExistException(OperationConstant.FAILED);
     }
 
+    @Override
+    public List<FriendDto> sortFriendsByLocation(FriendSearchDto friendSearchDto) {
+        Double accountLatitude = friendSearchDto.getLatitude();
+        Double accountLongitude = friendSearchDto.getLongitude();
+
+        Account loggedAccount = getLoggedAccount();
+        Set<Account> friends = loggedAccount.getFriends();
+
+        Comparator<Account> comparator = Comparator.comparingDouble(
+            friend -> {
+                if (friend.getLocation() == null) {
+                    return Double.MAX_VALUE;
+                }
+
+                Location friendLocation = friend.getLocation();
+                Double friendLongitude = friendLocation.getLongitude();
+                Double friendLatitude = friendLocation.getLatitude();
+
+                return calculateDistance(
+                    accountLatitude, accountLongitude, friendLatitude, friendLongitude);
+            }
+        );
+
+        List<Account> sortedFriends = new ArrayList<>(friends);
+        sortedFriends.sort(comparator);
+
+        return friendMapper.accountListToFriendDtoList(new ArrayList<>(sortedFriends));
```

Comment:
Why don't you pass sortedFriends without wrapping it again in ArrayList?

Reply:
Fixed